src/deskolem.ml

```diff
@@ -0,0 +1,49 @@
+open Core
+open Terms
+open Timed
+
+(** [subst_inv t s x] replaces the term starting with the symbol [s] by a fresh
+    variable [x] in the term [t]. *)
+let rec subst_inv : term -> sym -> term Bindlib.var -> term =
+  fun t s x ->
+  let h, ts = Basics.get_args t in
```

Comment:
add: `let ts = List.map subst ts in` here so that you don't need to write it again later.